components/nearby-diagram.js

```diff
@@ -201,35 +201,46 @@ export default function NearbyDiagram({
       index += 1;
     }
   });
+  let needNewScale = false;
   let scaleForVariantInLd = null;
   let offsetForVariantInLd = null;
   let VariantsInLdPositionY = null;
   let zoomInIconPositionForVariantsInLd = null;
   let scaleForVariantsInLdPositionY = null;
   if (uniqueVariantLD.length > 0) {
-    const variantsInLdRegionStart = uniqueVariantLD[0].start;
-    const variantsInLdRegionEnd =
-      uniqueVariantLD[uniqueVariantLD.length - 1].start;
-    const variantsInLdRegionMid =
-      (variantsInLdRegionEnd + variantsInLdRegionStart) / 2;
-    scaleForVariantInLd =
-      variantsInLdRegionMid > targetCoordinatesStart
-        ? viewBoxLength / 2 / (variantsInLdRegionEnd - targetCoordinatesStart)
-        : viewBoxLength /
-          2 /
-          (targetCoordinatesStart - variantsInLdRegionStart);
-    offsetForVariantInLd =
-      variantsInLdRegionMid < targetCoordinatesStart
-        ? variantsInLdRegionStart
-        : targetCoordinatesStart -
-          (variantsInLdRegionEnd - targetCoordinatesStart);
-    zoomInIconPositionForVariantsInLd =
-      regRegionPositionY +
-      Object.keys(regulatoryRegionsBySource).length *
-        (regRegionPositiontrackHeight + labelHeight);
-    scaleForVariantsInLdPositionY =
-      zoomInIconPositionForVariantsInLd + zoomInIconHeigt + 30 + blankHeight;
-    VariantsInLdPositionY = scaleForVariantsInLdPositionY + blankHeight;
+    needNewScale = needScaleForVariantLd(uniqueVariantLD, scaleForGene);
+    if (needNewScale) {
+      const variantsInLdRegionStart = uniqueVariantLD[0].start;
+      const variantsInLdRegionEnd =
+        uniqueVariantLD[uniqueVariantLD.length - 1].start;
+      const variantsInLdRegionMid =
+        (variantsInLdRegionEnd + variantsInLdRegionStart) / 2;
+      scaleForVariantInLd =
+        variantsInLdRegionMid > targetCoordinatesStart
+          ? viewBoxLength / 2 / (variantsInLdRegionEnd - targetCoordinatesStart)
+          : viewBoxLength /
+            2 /
+            (targetCoordinatesStart - variantsInLdRegionStart);
+      offsetForVariantInLd =
+        variantsInLdRegionMid < targetCoordinatesStart
+          ? variantsInLdRegionStart
+          : targetCoordinatesStart -
+            (variantsInLdRegionEnd - targetCoordinatesStart);
+      zoomInIconPositionForVariantsInLd =
+        regRegionPositionY +
+        Object.keys(regulatoryRegionsBySource).length *
+          (regRegionPositiontrackHeight + labelHeight);
+      scaleForVariantsInLdPositionY =
+        zoomInIconPositionForVariantsInLd + zoomInIconHeigt + 30 + blankHeight;
+      VariantsInLdPositionY = scaleForVariantsInLdPositionY + blankHeight;
+    } else {
+      scaleForVariantInLd = scaleForGene;
+      offsetForVariantInLd = offsetXForGene;
+      VariantsInLdPositionY =
+        regRegionPositionY +
+        Object.keys(regulatoryRegionsBySource).length *
+          (regRegionPositiontrackHeight + labelHeight);
+    }
```

Comment:
Sort of odd to have an upper-case variable that’s not a React component. Any reason it gets special treatment?

Reply:
I will rename it. Thank for spotting it!